src/Definition/HTMLDefinition.hack

```diff
@@ -258,7 +258,17 @@ class HTMLPurifier_HTMLDefinition extends HTMLPurifier\HTMLPurifier_Definition {
             "target" => new HTML\HTMLPurifier_AttrDef_HTML_FrameTarget(vec["_blank"]),
             "href" => new AttrDef\HTMLPurifier_AttrDef_URI()
         ];
-        $a_element = new HTMLPurifier\HTMLPurifier_ElementDef(true, $a_add_attr, vec[], vec[], vec[], new ChildDef\HTMLPurifier_ChildDef_Optional(),
+		$a_element = new HTMLPurifier\HTMLPurifier_ElementDef(
+			true,
+			$a_add_attr,
+			vec[],
+			vec[],
+			vec[
+				new HTMLPurifier\AttrTransform\HTMLPurifier_AttrTransform_TargetBlank(),
```

Comment:
I don't think we want this one to be applied all the time. It should only be added if the the config for `TargetBlank` is set to `true` and we have it set to `false` (the default) in both the service and the hack lib.

Unfortunately due to how we construct the HTMLDefinition.. I don't think we have access to the Config at this point (hard to tell since Config is pretty complex). What do you think about adding a check like
```php
if (!$config->get('HTML.TargetBlank')) {
  return $attr;
}
```

for these three new transforms?

Reply:
Hmm, but if these `AttrTransform` classes are not going to be constructed here, we need to find a new place to construct these. This is the place where they are called: https://github.com/slackhq/htmlsanitizer-hack/blob/main/src/AttrValidator.hack#L168-L176